lib/ansible/modules/storage/emc/emc_vnx_sg_member.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/python
+#
+# (c) 2018, Luca 'remix_tj' Lorenzetto <lorenzetto.luca@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: emc_vnx_sg_member
+
+short_description: Manage storage group member
+
+version_added: "2.7"
+
+description:
+    - "This module manages the members of an existing storage group"
+
+extends_documentation_fragment:
+  - emc.emc_vnx
+
+options:
+    name:
+        description:
+            - Name of the Storage group to manage
+        required: true
+    lunid:
+        description:
+            - Lun id to be added
+        required: true
+    state:
+        description:
+        - Indicates the desired lunid state.
+        - C(present) ensures specified lunid is present in the Storage Group.
+        - C(absent) ensures specified lunid is absent from Storage Group.
+        default: present
+        choices: [ "present", "absent"]
+
+
+author:
+    - Luca 'remix_tj' Lorenzetto (@remixtj)
+'''
+
+EXAMPLES = '''
+- name: Add lun to storage group
+  emc_vnx_sg_member:
+    name: sg01
+    sp_address: sp1a.fqdn
+    sp_user: sysadmin
+    sp_password: sysadmin
+    lunid: 100
+    state: present
+
+- name: Remove lun from storage group
+  emc_vnx_sg_member:
+    name: sg01
+    sp_address: sp1a.fqdn
+    sp_user: sysadmin
+    sp_password: sysadmin
+    lunid: 100
+    state: absent
+'''
+
+RETURN = '''
+hluid:
+    description: LUNID that hosts attached to the storage group will see
+    type: int
+    returned: success
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from storops import VNXSystem
+    from storops.exception import VNXCredentialError, VNXStorageGroupError, \
+        VNXAluAlreadyAttachedError, VNXAttachAluError, VNXDetachAluNotFoundError
+    HAS_LIB = True
+except:
+    HAS_LIB = False
+
+
+def run_module():
+    module_args = dict(
+        name=dict(type='str', required=True),
+        sp_address=dict(type='str', required=True),
+        sp_user=dict(type='str', required=False, default='sysadmin'),
+        sp_password=dict(type='str', required=False, default='sysadmin'),
```

Comment:
add `no_log=True` as this is password for something, which you might want to hide in logs and console logs.

Reply:
I'm worried i'll hit #19278 if the username and password are the same. This is not an unlikely event since sysadmin/sysadmin are the default credentials. Can you confirm there will be no such risk?

Reply:
I'm adding no_log anyway.

Reply:
This is known limitation with `no_log`, it won't affect the module functionality. It will just obfuscate password like words in log output.